incubator/etcd/templates/statefulset.yaml

```diff
@@ -6,47 +6,47 @@ metadata:
     heritage: {{ .Release.Service | quote }}
     release: {{ .Release.Name | quote }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    component: "{{ .Release.Name }}-{{ .Values.Component }}"
+    app: {{ template "etcd.name" . }}
 spec:
   serviceName: {{ template "etcd.fullname" . }}
-  replicas: {{ default 3 .Values.Replicas }}
+  replicas: {{ default 3 .Values.replicas }}
   template:
     metadata:
       name: {{ template "etcd.fullname" . }}
       labels:
         heritage: {{ .Release.Service | quote }}
         release: {{ .Release.Name | quote }}
         chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-        component: "{{ .Release.Name }}-{{ .Values.Component }}"
+        app: {{ template "etcd.name" . }}
     spec:
-{{- if .Values.Affinity }}
+{{- if .Values.affinity }}
       affinity:
-{{ toYaml .Values.Affinity | indent 8 }}
+{{ toYaml .Values.affinity | indent 8 }}
 {{- end }}
-{{- if .Values.NodeSelector }}
+{{- if .Values.nodeSelector }}
       nodeSelector:
-{{ toYaml .Values.NodeSelector | indent 8 }}
+{{ toYaml .Values.nodeSelector | indent 8 }}
 {{- end }}
-{{- if .Values.Tolerations }}
+{{- if .Values.tolerations }}
       tolerations:
-{{ toYaml .Values.Tolerations | indent 8 }}
+{{ toYaml .Values.tolerations | indent 8 }}
 {{- end }}
       containers:
       - name: {{ template "etcd.fullname" . }}
-        image: "{{ .Values.Image }}:{{ .Values.ImageTag }}"
-        imagePullPolicy: "{{ .Values.ImagePullPolicy }}"
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: "{{ .Values.image.pullPolicy }}"
         ports:
-        - containerPort: {{ .Values.PeerPort }}
+        - containerPort: {{ .Values.peerPort }}
           name: peer
-        - containerPort: {{ .Values.ClientPort }}
+        - containerPort: {{ .Values.clientPort }}
           name: client
         resources:
           requests:
```

Comment:
Don't specify defaults for resources. Just do this:

`{{ toYaml .Values.resources | indent 10 }}`